tests/test_module_loader.py

```diff
@@ -526,3 +527,77 @@ def test_module_has_v2_requirements_on_non_imported_module(snippetcompiler, loca
     )
     project.load_module_recursive()
     assert "minimalv2module" not in project.modules
+
+
+@pytest.mark.slowtest
+def test_project_requirements_dont_overwrite_core_requirements_source(
+    snippetcompiler_clean,
+    local_module_package_index: str,
+    modules_v2_dir: str,
+    tmpdir: py.path.local,
+) -> None:
+    """
+    A project has a requirement that is also a requirement of core
+    but with another version. The requirements of core should not be
+    overwritten. The module gets installed from source
+    """
+    # Create the module
+    module_name: str = "minimalv2module"
+    module_path: str = str(tmpdir.join(module_name))
+    module_from_template(
+        os.path.join(modules_v2_dir, module_name), module_path, new_requirements=[Requirement.parse("Jinja2==2.11.3")]
+    )
+
+    # Activate the snippetcompiler venv
+    project: Project = snippetcompiler_clean.setup_for_snippet("")
+    active_env = project.virtualenv
+    Jinja2_version_before = active_env.get_installed_packages()["Jinja2"].base_version
+
+    # Install the module
+    with pytest.raises(InvalidModuleException):
+        ModuleTool().install(editable=False, path=module_path)
+    Jinja2_version_after = active_env.get_installed_packages()["Jinja2"].base_version
+    assert Jinja2_version_before == Jinja2_version_after
```

Comment:
```suggestion
    jinja2_version_before = active_env.get_installed_packages()["Jinja2"].base_version

    # Install the module
    with pytest.raises(InvalidModuleException):
        ModuleTool().install(editable=False, path=module_path)
    jinja2_version_after = active_env.get_installed_packages()["Jinja2"].base_version
    assert jinja2_version_before == jinja2_version_after
```
Nitpicking here but I would stick to the Python naming conventions.